src/LSST/system/TelemetryCom.cpp

```diff
@@ -0,0 +1,272 @@
+    if (status < 0) {
+        LERROR("TelemetryCom::_clientConnect() connect failed host=", hostId, " port=", _port,
+               " status=", status);
+        return -1;
+    }
+    return clientFd;
+}
+
+bool TelemetryCom::waitForServerRunning(int seconds) {
+    for (int j = 0; j < seconds && !_serverRunning; ++j) {
+        sleep(1);
+    }
+    return _serverRunning;
+}
+
+void TelemetryCom::startServer() {
+    thread servThrd(&TelemetryCom::_server, this);
+    _serverThread = std::move(servThrd);
+}
+
+void TelemetryCom::_server() {
+    if (_shutdownComCalled) {
+        LERROR("TelemetryCom::_server() attempt to start after shutdown has been called");
+        return;
+    }
+
+    // Listen for new connections to accept
+    if ((_serverFd = socket(AF_INET, SOCK_STREAM, 0)) < 0) {
+        throw util::Bug(ERR_LOC, "TelemetryCom::server() failed to create listening socket");
+    }
+    int opt = 1;
+    if (setsockopt(_serverFd, SOL_SOCKET, SO_REUSEADDR | SO_REUSEPORT, &opt, sizeof(opt))) {
+        throw util::Bug(ERR_LOC, "TelemetryCom::server() failed to setsockopt");
+    }
+
+    struct sockaddr_in address;
+    int addrlen = sizeof(address);
+    address.sin_family = AF_INET;
+    address.sin_addr.s_addr = INADDR_ANY;
+    address.sin_port = htons(_port);
+
+    if (bind(_serverFd, (struct sockaddr*)&address, sizeof(address)) < 0) {
+        throw util::Bug(ERR_LOC, "TelemetryCom::server() failed to bind " + to_string(_port));
+    }
+    if (listen(_serverFd, 3) < 0) {
+        throw util::Bug(ERR_LOC, "TelemetryCom::server() failed to listen " + to_string(_port));
+    }
+    LINFO("TelemetryCom::server() listening");
+
+    _serverRunning = true;
+    while (_acceptLoop) {
+        int sock = accept(_serverFd, (struct sockaddr*)&address, (socklen_t*)&addrlen);
+        if (!_acceptLoop) {
+            continue;
+        }
+        if (sock < 0) {
+            LERROR("TelemetryCom::server() failed to accept on ", _port, " sock=", sock);
+            _acceptLoop = false;
+            continue;
+        }
+        // Create an object to handle the new connection.
+        LINFO("TelemetryCom::server() accepting new client");
+        {
+            auto handlerThrd =
+                    ServerConnectionHandler::Ptr(new ServerConnectionHandler(sock, _telemetryMap->copyMap()));
+            lock_guard<mutex> htLock();
+            _handlerThreads.push_back(handlerThrd);
+            // Check if any of the threads should be joined and removed.
+            auto iter = _handlerThreads.begin();
+            while (iter != _handlerThreads.end()) {
+                auto needToErase = (*iter)->checkJoin();
+                if (needToErase || (*iter)->getJoined()) {
+                    LINFO("TelemetryCom::server() removing old client");
+                    auto old = iter++;
+                    _handlerThreads.erase(old);
+                } else {
+                    ++iter;
+                }
+            }
```

Comment:
iter is incremented once every loop, so that should be ok. It can be made so increment only occurs on one line, which is probably safer in the case of future modification.